JavaScript/lib/avc278/linkedlist.js

```diff
@@ -0,0 +1,43 @@
+const assert = require("assert");
+
+class LinkedListNode {
+  constructor(val, next) {
+    this.val = val === undefined ? null : val;
+    this.next = next === undefined ? null : next;
+  }
+}
+
+const arrayToLinkedList = (arr) => {
+  let tail = null;
+  for (let i = arr.length - 1; i >= 0; i--) {
+    tail = new LinkedListNode(arr[i], tail);
+  }
+  return tail;
+};
+
+const compareLinkedLists = (A, B) => {
+  if (!A && !B) return true;
+
+  while (A !== null && B !== null) {
+    if (A.val !== B.val) return false;
+    A = A.next;
+    B = B.next;
+  }
+
+  return !((A && !B) || (!A && B));
```

Comment:
```suggestion
  return !A && !B;
```

Reply:
omg...
that's what I get for blindly c/p